lib/contents.ml

```diff
@@ -122,10 +122,47 @@ let readme config ppf =
 ```
 opam pin add --yes https://github.com/%s/%s.git
 opam install %s
-```|}
+```
+
+If you want to contribute or hack on the project, please read
+[CONTRIBUTING.md](CONTRIBUTING.md).|}
     config.project config.project_synopsis config.github_organisation
     config.project config.project
 
+let contributing config ppf =
+  Fmt.pf ppf
+    {|## Setting up your working environment
+
+%s requires OCaml %s or higher so you will need a corresponding opam switch
+which you can install by running:
+```
+$ opam switch create %s ocaml-base-compiler.%s
+```
+|}
+    config.project config.version_ocaml
```

Comment:
I named this variable poorly and didn't clean it up since it's not currently being used by any of the layouts (it should probably be `ocaml_version_min`). If we're going to use it here, we should set the default to something more ecosystem-friendly like `4.05.0`. However, as said above I'd prefer recommending `4.09` anyway.

Reply:
Should it be controlled by a max version instead?

I'm happy to hardcode 4.09 if that's what you prefer but the two things I considered were:
- if you set ocaml-min-version then the generated instructions will comply to your projects requirements, if it's hardcoded then that goes away
- suggesting the lowest supported version makes for a poorer dev experience but when you submit a PR it won't be rejected by the CI because you used a new feature that's not available to the whole range of supported compiler versions for the project.

Once again, I don't have strong opinions on this so let me know what you prefer!

Reply:
> if you set ocaml-min-version then the generated instructions will comply to your projects requirements, if it's hardcoded then that goes away

True, but we could have an `ocaml_version_max` too :slightly_smiling_face: I'm fairly happy to expect support for the latest release, I think. 

We could avoid hardcoding directly with a dependency on https://github.com/ocurrent/ocaml-version (which I was planning to do at some point).

> suggesting the lowest supported version makes for a poorer dev experience but when you submit a PR it won't be rejected by the CI because you used a new feature that's not available to the whole range of supported compiler versions for the project.

Good point. I would agree with you if the older syntax errors weren't so terrible.

Reply:
Makes sense, let's hardcode `4.09.0` for now, I'm happy to later improve it with a `--ocaml-version-max` or https://github.com/ocurrent/ocaml-version!

Reply:
I did just that, see the last commit. Let me know what you think!